hub/saml-proxy/src/main/java/uk/gov/ida/hub/samlproxy/exceptions/ExceptionAuditor.java

```diff
@@ -25,7 +25,7 @@ public boolean auditException(ApplicationException exception, Optional<SessionId
         if (!isAudited && exception.requiresAuditing()) {
             EventDetails eventDetails = new EventDetails(
                     downstream_uri,
-                    exception.getUri().or(URI.create("uri-not-present")).toASCIIString());
```

Comment:
Why the change to `orElse`?  Doesn't `or` or `orElseGet` only create the URI (invoke the supplier) if the Optional is empty? Whereas `orElse` will always create the URI. I know I like to say don't prematurely optimise but this would seem to be a case where it's better stick with a thing that does as little as possible?

Reply:
This is because saml-libs was switched from Guava Optionals to Java Optionals. The equivalent of Guava's `or` is `orElse`

Reply:
Oh, I missed the change of import in the commit - it looked like it had been the native Optionals for a while ...